File: 2160.拆分数位后四位数字的最小和.js
/*
 * @lc app=leetcode.cn id=2160 lang=javascript
 *
 * [2160] 拆分数位后四位数字的最小和
 */

// @lc code=start
/**
 * @param {number} num
 * @return {number}
 */
var minimumSum = function(num) {
    const digits = [];
    for (let i = 0; i < 4; i++) {
        digits.push(num % 10);
        num = Math.floor(num / 10);
    }
    digits.sort((a, b) => a - b);
    return (digits[0] + digits[1]) * 10 + (digits[2] + digits[3]);
};
// @lc code=end

File: 2160.拆分数位后四位数字的最小和.java
/*
 * @lc app=leetcode.cn id=2160 lang=java
 *
 * [2160] 拆分数位后四位数字的最小和
 */

// @lc code=start
class Solution {
    public int minimumSum(int num) {
        // 明显组成两个两位数,且最小的两个数字作为它们的十位数
        // 是最小的
        int[] digits = new int[4];
        for (int i = 3; i >= 0; i--) {
            digits[i] = num % 10;
            num /= 10;
        }
        Arrays.sort(digits);
        return (digits[0] + digits[1]) * 10 + (digits[2] + digits[3]);
    }
}
// @lc code=end

